File: app/src/main/java/org/ametro/ui/navigation/adapter/NavigationCheckBoxItemHolderFactory.java
package org.ametro.ui.navigation.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import org.ametro.R;

class NavigationCheckBoxItemHolderFactory implements IHolderFactory{

    @Override
    public IHolder createHolder(View convertView) {
        NavigationCheckBoxItemHolder holder = new NavigationCheckBoxItemHolder(convertView);
        convertView.setTag(holder);
        return holder;
    }

    @Override
    public View createView(LayoutInflater inflater, ViewGroup parent) {
        return inflater.inflate(R.layout.drawer_checkbox_item, parent, false);
    }

}

File: app/src/main/java/io/github/romangolovanov/apps/ametro/ui/navigation/adapter/NavigationCheckBoxItemHolderFactory.kt
package io.github.romangolovanov.apps.ametro.ui.navigation.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import io.github.romangolovanov.apps.ametro.R

internal class NavigationCheckBoxItemHolderFactory : IHolderFactory {

    override fun createHolder(view: View): IHolder {
        val holder = NavigationCheckBoxItemHolder(view)
        view.tag = holder
        return holder
    }

    override fun createView(inflater: LayoutInflater, parent: ViewGroup): View {
        return inflater.inflate(R.layout.drawer_checkbox_item, parent, false)
    }
}
